core/state.go

```diff
@@ -400,12 +400,16 @@ func (s *State) updateContractStorages(stateTrie *trie.Trie, diffs map[felt.Felt
 	})
 
 	// update per-contract storage Tries concurrently
-	contractUpdaters := pool.NewWithResults[*db.BufferedTransaction]().WithErrors().WithMaxGoroutines(runtime.GOMAXPROCS(0))
+	contractUpdaters := pool.NewWithResults[*db.BufferedTransactionWithAddress]().WithErrors().WithMaxGoroutines(runtime.GOMAXPROCS(0))
 	for _, key := range keys {
-		conractAddr := key
-		updateDiff := diffs[conractAddr]
-		contractUpdaters.Go(func() (*db.BufferedTransaction, error) {
-			return s.updateStorageBuffered(&conractAddr, updateDiff, blockNumber, logChanges)
+		contractAddr := key
+		updateDiff := diffs[contractAddr]
```

Comment:
Since this is only used once in the function there is no need for `updateDiff`, also since you are already making a shallow copy of the key in contractAddr it should be fine to have `diffs[contractAddr]` as the argument to `s.updatedStorageBuffered`.